pkg/snapstore/gcs_snapstore.go

```diff
@@ -62,15 +64,20 @@ func NewGCSSnapStore(config *brtypes.SnapstoreConfig) (*GCSSnapStore, error) {
 	emulatorConfig.enabled = isEmulatorEnabled()
 	var opts []option.ClientOption // no need to explicitly set store credentials here since the Google SDK picks it up from the standard environment variable
 
-	if _, ok := os.LookupEnv(envSourceStoreCredentials); !ok || emulatorConfig.enabled { // do not set endpoint override when copying backups between buckets, since the buckets may reside on different regions
-		endpoint := strings.TrimSpace(os.Getenv(envStorageAPIEndpoint))
+	if gcsApplicationCredentialsPath, isSet := os.LookupEnv(getEnvPrefixString(config.IsSource) + envStoreCredentials); isSet {
+		endpointFilePath := path.Join(path.Dir(gcsApplicationCredentialsPath), storageAPIEndpointFileName)
```

Comment:
Not clear on which file are we referring to here and why is it called `endpointFilePath`, can you please explain?

Reply:
Addressed in https://github.com/gardener/etcd-backup-restore/pull/759/commits/00e7d52969652e0390590750da06582ce6c336f5